packages/web-frontend/src/containers/Map/AreaTooltip.js

```diff
@@ -71,22 +66,28 @@ export class AreaTooltip extends Component {
       return Object.fromEntries(metadataKeys.map(k => [k.replace(`${key}_metadata`, ''), data[k]]));
     };
 
-    const formattedMeasureData = {};
-
-    if (orgUnitMeasureData) {
-      measureOptions.forEach(({ key, name, ...otherConfigs }) => {
-        const metadata = getMetadata(orgUnitMeasureData, key);
-        formattedMeasureData[name || key] = getSingleFormattedValue(orgUnitMeasureData, [
-          {
-            key,
-            metadata,
-            ...otherConfigs,
-          },
-        ]);
-      });
-    }
-
-    return formattedMeasureData;
+    const popUpList = orgUnitMeasureData
+      ? measureOptions
+          .filter(m => !m.hideFromPopup)
+          .map(({ key, name, values, ...otherConfigs }) => {
+            const metadata = getMetadata(orgUnitMeasureData, key);
+            const { formattedValue, valueInfo } = getFormattedInfo(orgUnitMeasureData, {
+              key,
+              metadata,
+              ...otherConfigs,
+            });
+            const formattedKey = name || key;
+            return valueInfo?.hideFromPopup ? null : (
+              <span key={key}>{`${formattedKey}: ${formattedValue}`}</span>
+            );
+          })
```

Comment:
I see a lot of commonality between this function and the one in `MeasurePopup.js`, think it's worth refactoring so that they share the same code? I suspect that the MeasurePopup would benefit from having this `getMetadata()` logic?

Reply:
@rohan-bes That's a good point, will refactor the code to share the function.👍